packages/react/scroll-area/src/useStateMachine.ts

```diff
@@ -0,0 +1,20 @@
+import * as React from 'react';
```

Comment:
Direct copy/paste of the state machine in `Presence`. Benoit and I spoke about copy/pasting things 3 times before we think about how to abstract them to avoid abstracting too early.